pkg/agent/openflow/multicast.go

```diff
@@ -99,6 +100,49 @@ func (f *featureMulticast) multicastOutputFlow(cookieID uint64) binding.Flow {
 		Done()
 }
 
+// multicastIngressIGMPPodMetricFlow generates the flow that matches IGMP query.
+func (f *featureMulticast) multicastIngressIGMPPodMetricFlow() binding.Flow {
+	return MulticastIngressPodMetricTable.ofTable.BuildFlow(priorityHigh).
+		Cookie(f.cookieAllocator.Request(f.category).Raw()).
+		MatchProtocol(binding.ProtocolIGMP).
+		MatchDstIP(types.McastAllHosts).
+		Action().NextTable().
+		Done()
+}
+
+// multicastEgressIGMPPodMetricFlow generates the flow that matches IGMP report.
+func (f *featureMulticast) multicastEgressIGMPPodMetricFlow() binding.Flow {
+	return MulticastEgressPodMetricTable.ofTable.BuildFlow(priorityHigh).
+		Cookie(f.cookieAllocator.Request(f.category).Raw()).
+		MatchProtocol(binding.ProtocolIGMP).
+		MatchDstIPNet(*types.McastCIDR).
+		Action().NextTable().
+		Done()
+}
+
+func (f *featureMulticast) multicastPodMetricFlows(podIP net.IP, podOFPort uint32) []binding.Flow {
+	ipProtocol := getIPProtocol(podIP)
+	return []binding.Flow{
+		// Generates the flows to forward multicast egress packets before outputting to MulticastOutputTable.
+		// It matches source IP with IP of the local sender Pod.
+		MulticastEgressPodMetricTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(f.cookieAllocator.Request(f.category).Raw()).
+			MatchProtocol(ipProtocol).
+			MatchSrcIP(podIP).
+			MatchDstIPNet(*types.McastCIDR).
```

Comment:
Not necessary for matching `*types.McastCIDR` in multicast pipeline if you do not specify a single IP or sub CIDR, since only the packet matches this CIDR would enter this pipeline.

Reply:
I am thinking this one's `MatchDstIPNet` should be removed as well?

```
func (f *featureMulticast) externalMulticastReceiverFlow() binding.Flow {
	return MulticastRoutingTable.ofTable.BuildFlow(priorityLow).
		Cookie(f.cookieAllocator.Request(f.category).Raw()).
		MatchProtocol(binding.ProtocolIP).
		MatchDstIPNet(*types.McastCIDR).
		Action().LoadRegMark(OFPortFoundRegMark).
		Action().LoadToRegField(TargetOFPortField, config.HostGatewayOFPort).
		Action().GotoStage(stageOutput).
		Done()
}
```

Reply:
Yes, it looks not necessary.

Reply:
updated. Removed unnecessary MatchDstIPNet(*types.McastCIDR) in multicast pipeline